eng/common/pipelines/templates/steps/sparse-checkout.yml

```diff
@@ -0,0 +1,20 @@
+parameters:
+  - name: Paths
+    type: object
+    default: []
+
+steps:
+  - checkout: none
+
+  - pwsh: |
+      git clone --no-checkout --filter=tree:0 git://github.com/$(Build.Repository.Name) .
```

Comment:
It is probably worth adding the repository as a template parameter as I believe this template will be interesting for pipelines we clone multiple pipelines. 

I also suggest adding a template parameter for the working directory for these steps so folks can control where this clone happens.

Reply:
I added a repositories object, since with multiple repos we need to pair the working directory with the repo name and commit-ish. Also, I set the default the way it is so that when no repo is specified, we just checkout into the default location, this way any calling template doesn't need to manually specify `WorkingDirectory` when there's only one repo.